src/components/grid-layout/cards/diagrams/networkAreaDiagram/network-area-diagram-content.tsx

```diff
@@ -129,6 +130,10 @@ function NetworkAreaDiagramContent(props: NetworkAreaDiagramContentProps) {
         []
     );
 
+    const handleToggleShowLabels = useCallback(() => {
+        setShowLabels(!showLabels);
+    }, [showLabels]);
```

Comment:
this is bad code. You will memoize your function `handleToggleShowLabels` at each usage
```suggestion
    const handleToggleShowLabels = useCallback(() => {
        setShowLabels((oldShowLabels) => !oldShowLabels);
    }, []);
```

Reply:
Thanks, fixed.